src/components/switch/switch-theme.scss

```diff
@@ -59,7 +59,72 @@ md-switch.md-THEME_NAME-theme {
       background-color: '{{background-400}}';
     }
     ._md-bar {
-      background-color: '{{foreground-4}}';
+      background-color: '{{background-contrast-0.12}}';
+    }
+  }
+
+  &.md-theme-dark {
+    .md-ink-ripple {
+      color: '{{background-contrast-0.30}}';
+    }
+    ._md-thumb {
+      background-color: '{{background-400}}';
+    }
+    ._md-bar {
+      background-color: '{{background-contrast-0.30}}';
+    }
+    &.md-checked {
+      .md-ink-ripple {
+        color: '{{accent-200}}';
+      }
+      ._md-thumb {
+        background-color: '{{accent-200}}';
+      }
+      ._md-bar {
+        background-color: '{{accent-200-0.5}}';
+      }
+      &.md-focused ._md-thumb:before {
+        background-color: '{{accent-200-0.26}}';
+      }
+
+      &.md-primary {
+        .md-ink-ripple {
+          color: '{{primary-200}}';
+        }
+        ._md-thumb {
+          background-color: '{{primary-200}}';
+        }
+        ._md-bar {
+          background-color: '{{primary-200-0.5}}';
+        }
+        &.md-focused ._md-thumb:before {
+          background-color: '{{primary-200-0.26}}';
+        }
+      }
+
+      &.md-warn {
+        .md-ink-ripple {
+          color: '{{warn-200}}';
+        }
+        ._md-thumb {
+          background-color: '{{warn-200}}';
+        }
+        ._md-bar {
+          background-color: '{{warn-200-0.5}}';
+        }
+        &.md-focused ._md-thumb:before {
+          background-color: '{{warn-200-0.26}}';
+        }
+      }
+    }
```

Comment:
This is is actually a repetition (3x). You should be able to create a local `mixin` and just specify the theme.